anta/tools/utils.py

```diff
@@ -32,3 +33,54 @@ def get_failed_logs(expected_output: dict[Any, Any], actual_output: dict[Any, An
             failed_logs.append(f"\nExpected `{expected_data}` as the {element}, but found `{actual_data}` instead.")
 
     return "".join(failed_logs)
+
+
+def custom_division(numerator: int | float, denominator: int | float) -> int | float:
+    """
+    Custom division that returns an integer if the result is an integer, otherwise a float.
+
+    Parameters:
+    numerator (float): The numerator.
+    denominator (float): The denominator.
+
+    Returns:
+    Union[int, float]: The result of the division.
+    """
+    result = numerator / denominator
+    return int(result) if result.is_integer() else result
+
+
+def extract_speed_and_lane(input_speed: str) -> tuple[Any, Any]:
+    """
+    This function extracts the speed and lane information from the input string.
+
+    Parameters:
+        input_speed (str): The input string which contains the speed and lane information.
+
+    Returns:
+        tuple[Any, Any]: The extracted speed from the input string, and the extracted lane from the input string.
+                         If no lane information is found, it returns None.
+
+    Examples:
+        100g-8: (100, 8)
+        100g: (100, None)
+        100-8: (100, 8)
```

Comment:
No, 100g-8 and 100g is the proper format but 100-8 is an extra option as we mentioned that input speed should be in gpbs so if user dont give `g` in input then also it will consider speed as gbps

Reply:
same lets not add oursevles things that are not on the device please